znvis/mesh/mesh.py

```diff
@@ -42,13 +42,15 @@ class Mesh:
 
     material: Material = field(default_factory=lambda: Material())
     base_direction: np.ndarray = field(default_factory=lambda: np.array([1, 0, 0]))
+    dynamic_color: bool = False
 
     def __post_init__(self):
         """
         Post init function to create materials.
         """
         material = rendering.MaterialRecord()
-        material.base_color = np.hstack((self.material.colour, self.material.alpha))
+        if not self.dynamic_color:
```

Comment:
Is the bool necessary? The color will either be a tuple or a nested array right? Can we not just check the dimension of the array and determine if it is time-dependent?